core/controllers/reader.py

```diff
@@ -597,6 +599,17 @@ def post(self, exploration_id):
             learner_progress_services.mark_collection_as_incomplete(
                 user_id, collection_id)
 
+        if user_id and story_id:
+            story = story_fetchers.get_story_by_id(story_id)
+            if story is not None:
+                learner_progress_services.mark_story_as_incomplete(
+                    user_id, story.id)
+                if story.corresponding_topic_id is not None:
+                    learner_progress_services.mark_topic_as_partially_learnt(
+                        user_id, story.corresponding_topic_id)
+            else:
+                raise Exception('Story with id %s does not exist.' % story_id)
```

Comment:
I don't think you need to raise an exception here. Maybe log it via logging.error? Since an error here would affect the user as well, but that is not necessary. (There are examples in the codebase on how to test logging.error as well)

Reply:
Done.